components/support/rs_client/src/error.rs

```diff
@@ -0,0 +1,17 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#[derive(Debug, thiserror::Error)]
+pub enum ClientError {
+    #[error("Error sending request: {0}")]
+    RequestError(#[from] viaduct::Error),
+    #[error("Error parsing URL: {0}")]
+    UrlParsingError(#[from] url::ParseError),
+    #[error("Server asked the client to back off ({0} seconds remaining)")]
+    BackoffError(u64),
+    #[error("Error in network response: {0}")]
+    ResponseError(String),
```

Comment:
Is there a reason you don't have this be a `viaduct::Response` instead of a string? That way consumers of the error can inspect it for more details if they care - eg, for the status code.

Reply:
I just copied it over from Nimbus tbh. I would guess they stringified it so that a consuming application could get details of the error which otherwise get lost since the `viaduct::Response` does not implement `Display`. 

Is there a good way to use `viaduct::Response` here and have the error use full display text from Nimbus and the top-level Remote Settings client? Otherwise I think it may be best to keep as is so the application layer can benefit fully.

Reply:
yeah, I later realized that this will be because it's exposed via uniffi, which a viaduct::Response can't do, so yeah, this is fine.